### ISSUE TITLE
Route constraints doesn't work in minimal Web API

### ISSUE THREAD

**Issue by Marusyk** [dotnet/aspnetcore#37109]
### Describe the bug

[Route constraints](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-6.0#route-constraint-reference) doesn't work in minimal Web API

The following endpoint returns 404
```
app.MapGet("/hello/{param:alpha}", (string param) => new { Hello = param });
```

![image](https://user-images.githubusercontent.com/9221694/135343679-b2433a37-293d-4226-902d-16d20b39cc58.png)

### Further technical details
- ASP.NET Core 6
- dotnet version:   6.0.100-preview.7.21379.14
- Microsoft Visual Studio Community 2022 Preview (64-bit)
Version 17.0.0 Preview 3.1

**Comment by martincostello** [comment]
Possible duplicate of #36413

**Comment by BrennanConroy** [comment]
Like Martin said, this has been fixed in swashbuckle 6.2.2 and we have some followup work to make it properly constrain the type on the UI end #36525
